### PR TITLE
Fix a bunch of ESLint and HTMLLint errors

### LINKED ISSUE #82: Duplicate `upload-img` id in views/index.handlebars
Ref: [/views/index.handlebars:18](https://github.com/mozilla/something-awesome/blob/8baa6d096461670fc395e5c68f02c413e273dacb/views/index.handlebars#L18) and [/views/index.handlebars:54](https://github.com/mozilla/something-awesome/blob/8baa6d096461670fc395e5c68f02c413e273dacb/views/index.handlebars#L54):

```html
<div id="upload-img"><img src="/resources/upload.svg" alt="Upload" /></div>
```

---

You can see this in person by running `npm run lint:html` from the CLI.

### Output:

```sh
...
> portal-alpha@1.0.0 lint:html /Users/pdehaan/dev/github/mozilla/something-awesome
> htmllint 'views/*.handlebars'

views/index.handlebars: line 54, col 13, the id "upload-img" is already in use

[htmllint] found 1 errors out of 2 files
```

### PR DESCRIPTION
**Pull request by pdehaan**
Also, try linting code on Circle-CI.

Fixes #82

### CODE REVIEW

File: test/aws.storage.test.js [hunk 1]
```diff
@@ -21,145 +21,156 @@ redisStub.createClient = function() {
...
 
   after(function() {
     crypto.randomBytes.restore();
-  })
+  });
 
   it('Should pass when the file is successfully uploaded and no bitly key', function() {
     conf.bitly_key = null;
-    const buf =  new Buffer(10);
+    const buf = Buffer.alloc(10);
```
Comment (pdehaan) [inline test/aws.storage.test.js]: Note that `new Buffer(<size>)` seems to be deprecated: https://nodejs.org/api/buffer.html#buffer_new_buffer_size

File: .eslintrc.yml [hunk 2]
```diff
@@ -16,18 +14,10 @@ plugins:
...
-  security/detect-disable-mustache-escape: warn
-  security/detect-eval-with-expression: warn
-  security/detect-new-buffer: warn
-  security/detect-no-csrf-before-method-override: warn
-  security/detect-non-literal-fs-filename: warn
-  security/detect-non-literal-regexp: warn
-  security/detect-non-literal-require: warn
-  security/detect-possible-timing-attacks: warn
-  security/detect-pseudoRandomBytes: warn
-  security/detect-unsafe-regex: warn
```
Comment (pdehaan) [inline .eslintrc.yml]: These are now all in the extended `"plugin:security/recommended"` ruleset.

File: package.json [hunk 3]
```diff
@@ -44,7 +45,7 @@
   "repository": "mozilla/something-awesome",
   "scripts": {
     "dev": "watchify frontend/src/main.js -o public/bundle.js -d | node server/portal_server.js",
-    "format": "prettier 'frontend/src/*.js' 'server/*.js' 'public/*.css' --single-quote --write",
+    "format": "prettier 'frontend/src/*.js' 'public/*.css' 'server/*.js' 'test/*.js' --single-quote --write",
```
Comment (pdehaan) [inline package.json]: `npm run format` will now format all the mocha test files in `test/*.js`.

File: test/aws.storage.test.js [hunk 4]
```diff
@@ -21,145 +21,156 @@ redisStub.createClient = function() {
...
 describe('Testing Delete from aws', function() {
   it('Returns successfully if the id is deleted off aws', function() {
     hget.callsArgWith(2, null, 'delete_token');
     s3Stub.deleteObject.callsArgWith(1, null, {});
-    return storage.delete('file_id', 'delete_token')
-                  .then(reply => assert(1), err => assert.fail())
-  })
+    return storage
+      .delete('file_id', 'delete_token')
+      .then(_reply => assert(1), err => assert.fail());
```
Comment (pdehaan) [inline test/aws.storage.test.js]: Wasn't sure how we felt about the `.then(yay, boo)` syntax versus converting it to `.then(yay).catch(boo)` style, so I left these as-is.

File: views/index.handlebars [hunk 5]
```diff
@@ -51,7 +51,9 @@
         Uploading
       </div>
       <div class="upload-window">
+        <!-- htmllint id-no-dup="false" -->
         <div id="upload-img"><img src="/resources/upload.svg" alt="Upload" /></div>
+        <!-- htmllint tag-bans="$previous" -->
```
Comment (pdehaan) [inline views/index.handlebars]: This should "fix" #82 by having htmllint ignore that rule for that block. Although refactoring to use classes or redoing the UI will also fix it for reals.

**Comment by dannycoates** [comment]
@pdehaan I gave you 'write' to the repo so you can push branches here so we can help fix conflicts easier.

**Comment by pdehaan** [comment]
@dannycoates Any ideas on why my PRs don't seem to be running on Circle-CI?

**Comment by dannycoates** [comment]
> Any ideas on why my PRs don't seem to be running on Circle-CI?

Builds from forks was disabled. I've enabled it

**Comment by pdehaan** [comment]
Yay! I failed!
Looks like **htmllint** really doesn't like Handlebars partials (Ref: https://github.com/htmllint/htmllint/issues/198).
Not sure how to proceed apart from removing htmllint support for the time being.

**Pull request merged**